packages/nvidia_nat_mcp/src/nat/plugins/mcp/client_base.py

```
@@ -181,30 +201,88 @@ def server_name(self):
+                try:
+                    # Close the existing stack and ClientSession
+                    if self._exit_stack:
+                        await self._exit_stack.aclose()
+                    # Create a fresh stack and session
+                    self._exit_stack = AsyncExitStack()
+                    self._session = await self._exit_stack.enter_async_context(self.connect_to_server())
+
+                    self._connection_established = True
+
+                    logger.info("Reconnected to MCP server (%s) on attempt %d", self.server_name, attempt)
+                    return
```

Ruff (0.13.1)

Consider moving this statement to an `else` block

(TRY300)